.github/workflows/ci.yml

```diff
@@ -279,3 +281,25 @@ jobs:
 
             If this is not what was expected, commit changes to `config/versions.json`.
             </details>
+
+      - name: PR Description Notifier
+        env:
+          GH_TOKEN: ${{ github.token }}
+          PR_BODY_FILE_LOCATION: ./body.txt
+          LATEST_PRERELEASE_SECTION_REGEX: "^:rocket: .* :rocket:$"
+          # We can't use many punctuation marks in the Prerelease Section because `sed` treats
+          # the replacement string as a regex string - no special regex chars without escaping
```

Comment:
What do you mean by punctuation marks?

I think as much as trying to avoid some characters can be safe, it might not be safe enough depending on the values of `${{ needs.defaults.outputs.root-sbd }}`, `${{ needs.check-spack-yaml.outputs.prerelease }}` and `${{ steps.comment.outputs.comment-link }}` (if any of that contains metacharacters, especially `&`, it might be a problem).
A safer solution might be escaping `sed`'s replacement string before using it within the `sed` command. 
In case you want to be safer, [this answer](https://stackoverflow.com/a/29613573/21024780) has extensive solutions for escaping sed strings.

Reply:
Yeah, fair enough. I just didn't want to have to do an escaped and non-escaped version of the `echo` variant :D I'll think up a solution...

Reply:
I think you could have something like:
```bash
LATEST_PRERELEASE_SECTION='${{ env.LATEST_PRERELEASE_SECTION }}'
ESCAPED_LATEST_PRERELEASE_SECTION= ... # command to escape the string
```
and then for your `sed` command use:
```bash
sed -i 's|${{ env.LATEST_PRERELEASE_SECTION_REGEX }}|`"$ESCAPED_LATEST_PRERELEASE_SECTION|g" ${{ env.PR_BODY_FILE_LOCATION }}
```
and keep `${{ env.LATEST_PRERELEASE_SECTION }}` (or you can also use `$LATEST_PRERELEASE_SECTION`) for the `echo` command.

Note you might also want to use `shell: bash +H {0}` to avoid problems with the `!` character (history expansion)

Reply:
I've managed to get around the escaping issue by using `awk` instead of `sed` - done in commit https://github.com/ACCESS-NRI/build-cd/pull/172/commits/3e65b4bfc90f16ab3b113eddbc9fa9ab436cd846

See the test runs for https://github.com/codegat-test-org/test/pull/22 in https://github.com/codegat-test-org/test/actions/runs/11964783047 and https://github.com/codegat-test-org/test/actions/runs/11964805872